src/tactic/simp_result.lean

```diff
@@ -0,0 +1,130 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Gabriel Ebner, Scott Morrison
+-/
+import tactic.core
+
+/-!
+# simp_result
+
+`dsimp_result` and `simp_result` are a pair of tactics for
+cleaning up the results produced by other tactics.
+
+They should be used with some caution.
+You should consider whether there is any real need for this clean-up,
+and whether there is a more direct way of producing the result you wanted,
+before relying on these tactics.
+
+Both are implemented in terms of a generic `intercept_result` tactic,
+which allows you to run an arbitrary tactic and modify the returned results.
+-/
+
+namespace tactic
+
+/--
+`intercept_result m t`
+attempts to run a tactic `t`,
+intercepts any results `t` assigns to the goals,
+and runs `m : expr → tactic expr` on each of the expressions
+before assigning the returned values to the original goals.
```

Comment:
Is this potentially unsafe, in that `m` could change the type of a solved goal? (I'm not 100% sure on the semantics of the `unsafe_type_context` functions.) It would be helpful to have a note in the doc string about this.

Reply:
I've added a warning that a badly behaved `m` could cause trouble, but to be honest I don't know how bad it could really get here!

Reply:
This should be safe, it's just a more reliable way to write `unify g g'`.